### PR TITLE
fixed anti affinity definition

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: helm/e2e-app-chart/templates/deployment.yaml [hunk 1]
```diff
@@ -18,13 +18,13 @@ spec:
...
-                    values:
-                    - e2e-app
-              topologyKey: "kubernetes.io/hostname"
+          - labelSelector:
+              matchExpressions:
+              - key: "app"
+                operator: In
+                values:
+                - e2e-app
+            topologyKey: "kubernetes.io/ip"
```
Comment (xh3b4sd) [inline helm/e2e-app-chart/templates/deployment.yaml:27]: This is the main thing that gets fixed. See https://gigantic.slack.com/archives/C12SG2929/p1523893101000222.

Comment (xh3b4sd) [reply helm/e2e-app-chart/templates/deployment.yaml:27]: Note I tested this on the guest cluster created in CI build https://circleci.com/gh/giantswarm/aws-operator/5433.

**Review by MarcelMue — APPROVED** [review]
LGTMIIWFY

We will have to fix this in other places too, right?

**Comment by xh3b4sd** [comment]
Lets discuss in the thread.

**Pull request merged**